src/ds_caselaw_utils/courts.py

```diff
@@ -30,12 +30,20 @@ def __init__(self, name, courts):
 class CourtsRepository:
     def __init__(self, data):
         self._data = data
+        self._byParam = {}
+        self._byCode = {}
+        for group in self._data:
+            for courtData in group.get("courts"):
```

Comment:
Nice optimisation; we need to be careful not to initialise the `CourtsRepository` in loops, but I don't think there's any case where we do that at the moment!